website/server/managers/LoginManager.js

```diff
@@ -1,10 +1,70 @@
+const Helper = require('../Helper.js');
+const config = require('../Config.json');
+
 // manages login logic and database queries
 class LoginManager
 {
     constructor (database)
     {
         this.database = database;
     }
+    // method adds a user to the database
+    addUser(name, password)
+    {
+        let hashedPassword = this.hash(password);
+        this.database.run(`INSERT INTO Users VALUES (NULL, ?, ?, 0);`, [name, hashedPassword])
+            .catch(Helper.log);
+    }
+    // method removes user from the database
+    removeUser(name)
+    {
+        this.database.run(`DELETE FROM Users WHERE name = ?;`, [name])
+            .catch(Helper.log);
+    }
+    // method check if a login info is valid. returns promise with user id or rejects if invalid.
+    loginValid(name, password)
+    {
+        let hashedPassword = this.hash(password);
+        const LOGIN_FAILED = "login failed", INCORRECT_PASSWORD = "incorrect password";
+        return this.database.select(`SELECT password, id FROM Users WHERE name = ?;`, [name])
+            .then(
+                data =>
+                {
+                    // check if exactly one user was found
+                    if (data.length === 1)
+                        data = data[0];
+                    else throw new Error(LOGIN_FAILED);
```

Comment:
please separate the cases, if the user doesn't exists it shouldn't be login failed. beside that very good :)